BigQuery connector for Hadoop 1.0.0 does not include repackaged dependencies

Until recently I had my Dataproc job configured to use the BigQuery connector for Hadoop by including this URL in my jarFileUris:

```
gs://hadoop-lib/bigquery/bigquery-connector-hadoop2-latest.jar
```

This last worked for me on 2019-08-18. When I ran my job on 2019-08-25, I got this error:

```
19/08/24 22:38:56 INFO hadoop.IntervalDataJob: IntervalDataJob: 01.26.00-SNAPSHOT built on 2019-07-15 06:56:09 AEST
19/08/24 22:38:56 INFO hadoop.BigQueryUtils: Using export bucket 'dataproc-f012147d-6848-4198-ad8b-8c5787e59478-us-central1' as specified in 'mapred.bq.gcs.bucket'
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.cloud.hadoop.services.agent.job.shim.HadoopRunClassShim.main(HadoopRunClassShim.java:19)
Caused by: java.lang.NoClassDefFoundError: com/google/cloud/hadoop/repackaged/bigquery/com/google/common/collect/ImmutableList
	at com.google.cloud.hadoop.io.bigquery.BigQueryConfiguration.<clinit>(BigQueryConfiguration.java:84)
	at au.com.redenergy.lebowski.hadoop.IntervalDataJob.main(IntervalDataJob.java:76)
	... 5 more
Caused by: java.lang.ClassNotFoundException: com.google.cloud.hadoop.repackaged.bigquery.com.google.common.collect.ImmutableList
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 7 more
Job output is complete
```

I changed the code to include the connector via an initialization action. After experimenting with connector versions, I found that versions 1.0.0 and 1.0.1-RC2 have this problem, but 0.13.17 does not. I was able to get my job to run by going back to connector version 0.13.17.

Looking in the source of `com.google.cloud.hadoop.io.bigquery.BigQueryConfiguration` provided in version hadoop2-1.0.0, it includes this import:

```
import com.google.common.collect.ImmutableList;
```

and this static field:

```
  public static final ImmutableList<String> MANDATORY_CONFIG_PROPERTIES_INPUT =
      ImmutableList.of(
          PROJECT_ID_KEY, INPUT_PROJECT_ID_KEY, INPUT_DATASET_ID_KEY, INPUT_TABLE_ID_KEY);
```

Therefore it is not possible to load this class unless `ImmutableList` is available.

Is this simply a bug or oversight, leaving this dependencies out?

Or should I be arranging to have the libraries available myself? 

Or is this class or Hadoop version deprecated and I should be changing my code?

Thanks for any feedback on this issue.

--- a/bigquery/pom.xml
+++ b/bigquery/pom.xml
@@ -209,12 +209,6 @@
                   </includes>
                 </filter>
                 <filter>
-                  <artifact>com.google.guava:*</artifact>
-                  <includes>
-                    <include>com/google/common/base/*</include>
-                  </includes>
-                </filter>
-                <filter>
                   <artifact>com.google.flogger:flogger-log4j-backend</artifact>
                   <includes>
                     <include>**</include>
@@ -239,8 +233,8 @@
                   <include>com.google.cloud</include>
                   <include>com.google.cloud.bigdataoss:util</include>
                   <include>com.google.cloud.bigdataoss:util-hadoop</include>
-                  <include>com.google.guava</include>
                   <include>com.google.flogger</include>
+                  <include>com.google.guava</include>
                   <include>com.google.http-client:google-http-client</include>
                   <include>com.google.http-client:google-http-client-jackson2</include>
                   <include>com.google.oauth-client:google-oauth-client</include>
@@ -266,7 +260,6 @@
                   </includes>
                   <excludes>
                     <exclude>com.google.cloud.hadoop.io.bigquery.**</exclude>
-                    <exclude>com.google.cloud.hadoop.repackaged.**</exclude>
                     <exclude>com.google.cloud.hadoop.util.AccessTokenProvider</exclude>
                     <exclude>com.google.cloud.hadoop.util.AccessTokenProvider$AccessToken</exclude>
                   </excludes>

